.eslintrc.js

```diff
@@ -337,13 +337,20 @@ module.exports = {
                   '!src/core/server/types',
                   '!src/core/server/*.test.mocks.ts',
 
-                  'src/plugins/**/public/**/*',
-                  '!src/plugins/**/public/index.{js,ts,tsx}',
+                  '(src|x-pack)/plugins/**/public/**/*',
+                  '!(src|x-pack)/plugins/**/public/index.{js,ts,tsx}',
 
-                  'src/plugins/**/server/**/*',
-                  '!src/plugins/**/server/index.{js,ts,tsx}',
+                  '(src|x-pack)/plugins/**/server/**/*',
+                  '!(src|x-pack)/plugins/**/server/index.{js,ts,tsx}',
                 ],
                 allowSameFolder: true,
+                errorMessage: 'Plugins may only import from top-level public and server modules.',
+              },
+              {
+                target: ['(src|x-pack)/plugins/**/*', '!(src|x-pack)/plugins/*/server/**/*'],
```

Comment:
Shouldn't it complain when a plugin reads from server/types? Probably we have it blacklisted, I'm from phone and cannot see

Reply:
Are you saying we should allow imports from `server/types` or not? In the current state of the PR it does not allow this import. I think we should instead, encourage plugins to move shared things into a `common` directory just like other code.

Reply:
agree